src/patterns/pattern/mod.rs

```diff
@@ -58,26 +97,25 @@ impl Pattern {
         if is_inside {
             let color = board.color(neighbour);
             match point_pattern {
-                'X' => { color == Black }
-                'O' => { color == White }
-                '?' => { true }
-                'x' => { color != Black }
-                'o' => { color != White }
-                '.' => { color == Empty }
-                ' ' => { false }
-                _  => { panic!("Unknown pattern {:?}", point_pattern) }
+                PointPattern::Black => { color == Black }
+                PointPattern::White => { color == White }
+                PointPattern::All => { true }
+                PointPattern::NotBlack => { color != Black }
+                PointPattern::NotWhite => { color != White }
+                PointPattern::Empty => { color == Empty }
+                PointPattern::OffBoard => { false }
             }
         } else {
-            point_pattern == ' '
+            point_pattern == PointPattern::OffBoard
         }
     }
 
-    fn point_pattern_for(&self, coord: &Coord, neighbour: &Coord) -> char {
+    fn point_pattern_for(&self, coord: &Coord, neighbour: &Coord) -> PointPattern {
```

Comment:
This is the second method that needs to be sped up. The best is of course to call it less, so that's what the next PR will be about.